src/tools/gcc.lua

```diff
@@ -236,8 +236,20 @@
 
 	gcc.libraryDirectories = {
 		architecture = {
-			x86 = "-L/usr/lib32",
-			x86_64 = "-L/usr/lib64",
+			x86 = function (cfg)
+				local r = {}
+				if cfg.system ~= premake.MACOSX then
+					table.insert (r, "-L/usr/lib32")
+				end
+				return r
+			end,
+			x86_64 = function (cfg)
+				local r = {}
+				if cfg.system ~= premake.MACOSX then
+					table.insert (r, "-L/usr/lib64")
+				end
+				return r
+			end,
```

Comment:
Sorry, I wasn't very clear about what I meant, what @noresources has makes more sense than what I was suggesting, given the examples they've provided. If @tvandijck is happy with this, then we should be good to merge it in as is.